pkg/chartverifier/checks/charttesting.go

```diff
@@ -102,6 +132,10 @@ func ChartTesting(opts *CheckOptions) (Result, error) {
 		}
 	}
 
+	if versionError := setOCVersion(opts, &version{}); versionError != nil {
+		return NewResult(false, versionError.Error()), nil
```

Comment:
Instead of abusing 'opts' to hold the AnnotationHolder value, I'd augment the `Result` type with a map to hold arbitrary values collected by the checks; this would effectively allow the caller to decide what to propagate upwards instead of delegating the decision to the check. IOW, `return NewResult(true, ChartTestingsuccess, map[string]string{}{"osVersion": osVersion})`.

Reply:
Not sure I agree this is abusing opts, maybe not a perfect fit but opts can be anything you need to pass to a function to enable it to work - it is our internal structure so we define its scope. I am not a fan of putting it in results either. I want to propose leaving as-is for now so we can create a release tomorrow and open an issue to figure out how best to do this.

> I particularly don't like the CheckOptions type to contain values such as the AnnotationHolder for a simple reason: tests should be treated as separated programs that do not interfere with the rest of the verification process. Like mentioned in another opportunity, I'd rather expose this as a structure downwards (as in adding another field, sibling of 'Reason') instead of this indirection. IOW, if the OpenShift version is important mainly for the chart-testing check, then let's remove it from the report metadata and remove this coupling.

I don't think we have broken the separation here, just consider that part of the testing check is whether we can get the OpenShift version and if we can't get it the check fails. The bad/difficult part is getting into an annotation given the structure of the code. it may have been better to have an extra check to to check we can get an OpenShift version but not at this stage. Anyway, as I say above, can we agree to leave as-is for now and open an issue to fix it after the release?